src/Kriti/Eval.hs

```diff
@@ -145,16 +179,16 @@ evalWith funcMap = \case
     t2' <- eval t2
     case (t1', t2') of
       (J.Bool p, J.Bool q) -> pure $ J.Bool $ p && q
-      (t1'', J.Bool _) -> throwError $ TypeError src sp $ renderBL $ "'" <> J.encode t1'' <> "' is not a boolean."
-      (_, t2'') -> throwError $ TypeError src sp $ renderBL $ "'" <> J.encode t2'' <> "' is not a boolean."
+      (json, J.Bool _) -> throwError $ TypeError src sp json "Boolean"
```

Comment:
These hard-coded "Booleans" make me think this could be improved, or at least a bool specific version of the function created.

Reply:
yeah i have some ideas for how to clean up the eval function.